unsafe/src/test/java/org/apache/spark/unsafe/types/IntervalSuite.java

```diff
@@ -56,4 +56,53 @@ public void toStringTest() {
     i = new Interval(34, 3 * MICROS_PER_WEEK + 13 * MICROS_PER_HOUR + 123);
     assertEquals(i.toString(), "interval 2 years 10 months 3 weeks 13 hours 123 microseconds");
   }
+
+  @Test
+  public void fromStringTest() {
+    testSingleUnit("year", 3, 36, 0);
+    testSingleUnit("month", 3, 3, 0);
+    testSingleUnit("week", 3, 0, 3 * MICROS_PER_WEEK);
+    testSingleUnit("day", 3, 0, 3 * MICROS_PER_DAY);
+    testSingleUnit("hour", 3, 0, 3 * MICROS_PER_HOUR);
+    testSingleUnit("minute", 3, 0, 3 * MICROS_PER_MINUTE);
+    testSingleUnit("second", 3, 0, 3 * MICROS_PER_SECOND);
+    testSingleUnit("millisecond", 3, 0, 3 * MICROS_PER_MILLI);
+    testSingleUnit("microsecond", 3, 0, 3);
+
+    String s;
+    Interval i;
+
+    s = "interval   -5  years  23   month";
+    i = new Interval(-5 * 12 + 23, 0);
+    assertEquals(Interval.fromString(s), i);
+
+    // Error cases
+    i = null;
```

Comment:
remove i and use null directly. 

"i" is a bad name since it is usually used for iterating over an array.